Libraries/Geolocation/RCTLocationObserver.m

```diff
@@ -148,9 +161,34 @@ - (dispatch_queue_t)methodQueue
 
 - (NSArray<NSString *> *)supportedEvents
 {
-  return @[@"geolocationDidChange", @"geolocationError"];
+  return @[@"geolocationDidChange",
+           @"geolocationError",
+           @"locationAuthorizationStatusDidChange"];
+}
+
+- (void)startObserving
+{
+  _hasListeners = YES;
+  if (!_locationManager) {
+    _locationManager = [CLLocationManager new];
+    _locationManager.delegate = self;
+  }
 }
 
+- (void)stopObserving
+{
+  _hasListeners = NO;
+}
+
+- (NSDictionary *)constantsToExport
```

Comment:
Why do we need this? Implementing `constantsToExport` prevents this module to be lazy-loaded which affects performance.

Reply:
Well to be able to compare them to the value of the `authorizationStatusChange`. So something like this:

```js
Geolocation.addEventListener('authorizationStatusChange', (status) => {
  if (Geolocation.authorizedAlways === status) {
    // ...
  }
  if (Geolocation.denied === status) {
    // ...
  }
});
```

Or am I missing something and can this be achieved in another way?